hal/src/nRF52840/ble_hal.cpp

```diff
@@ -1866,6 +1880,117 @@ int BleObject::ConnectionsManager::getConnectionInfo(hal_ble_conn_handle_t connH
     return SYSTEM_ERROR_NONE;
 }
 
+int BleObject::ConnectionsManager::setPairingConfig(const hal_ble_pairing_config_t* config) {
+    CHECK_TRUE(config, SYSTEM_ERROR_INVALID_ARGUMENT);
+    ioCaps_ = config->io_caps;
+    return SYSTEM_ERROR_NONE;
+}
+
+int BleObject::ConnectionsManager::startPairing(hal_ble_conn_handle_t connHandle) {
+    BleConnection* connection = fetchConnection(connHandle);
+    CHECK_TRUE(connection, SYSTEM_ERROR_NOT_FOUND);
+    CHECK_TRUE(connection->pairState == BLE_PAIRING_STATE_NOT_INITIATED ||
+               connection->pairState == BLE_PAIRING_STATE_REJECTED ||
+               connection->pairState == BLE_PAIRING_STATE_SEC_REQ_RECEIVED, SYSTEM_ERROR_INVALID_STATE);
+    ble_gap_sec_params_t secParams = {};
+    if (ioCaps_ != BLE_IO_CAPS_NONE) {
+        secParams.mitm = 1;
+    }
+    if (connection->info.role == BLE_ROLE_CENTRAL) {
+        secParams.io_caps        = toPlatformIoCaps(ioCaps_);
+        secParams.oob            = 0;
+        secParams.min_key_size   = 7;
+        secParams.max_key_size   = 16;
+        secParams.kdist_own.enc  = 1;
+        secParams.kdist_own.id   = 1;
+        secParams.kdist_peer.enc = 1;
+        secParams.kdist_peer.id  = 1;
```

Comment:
Nit: magic numbers

Reply:
For the one bit-filed member, I'll just use `true`/`false` instead.